java/ql/test/kotlin/library-tests/parameter-defaults/erasure.ql

```diff
@@ -5,6 +5,8 @@ class InstantiatedType extends ParameterizedType {
 }
 
 // This checks that all type parameter references are erased in the context of a $default function.
+// Note this is currently expected to fail since for the time being we extract type variable references
+// even where they should be out of scope.
```

Comment:
I don't think this is specific to `$default` functions. This applies in general: type variables should not be referenced if they are not in scope. Should this test be turned into a consistency query?

Reply:
Yes, I just haven't done the work yet :)